extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/sql/ThetaSketchBaseSqlAggregator.java

```diff
@@ -116,26 +121,38 @@ public Aggregation toDruidAggregation(
         );
       }
 
-      final DimensionSpec dimensionSpec;
-
-      if (columnArg.isDirectColumnAccess()) {
-        dimensionSpec = columnArg.getSimpleExtraction().toDimensionSpec(null, inputType);
-      } else {
-        String virtualColumnName = virtualColumnRegistry.getOrCreateVirtualColumnForExpression(
-            columnArg,
-            dataType
+      if (!inputType.is(ValueType.COMPLEX)) {
```

Comment:
return with error here

Reply:
Where? Do you mean the following?
```java
if (!inputType.is(ValueType.COMPLEX)) {
    // code
}
else {
      plannerContext.setPlanningError(xyz);
      return null;
}
```

If yes, this would run into the same duplication reasoning as the other comments: https://github.com/apache/druid/pull/16682#discussion_r1677509084

Reply:
I think yes ; you are trying to infer backward from `aggregator == null` that the issue was that the aggregation is not supported with this type - although that might be true right now ; that could change in the future - if for some reason in some other cases the `aggregate` will become `null`; but in that case the check will still hit - and produce a misleading error...

note: you don't need an `else` if there is a `return`
I don't think its beneficial to keep running till the function end if its known that the operation contract was already violated
```
if (inputType.is(ValueType.COMPLEX)) {
  plannerContext.setPlanningError(xyz);
  return null; 
}
 // code
```

Reply:
Made the change.